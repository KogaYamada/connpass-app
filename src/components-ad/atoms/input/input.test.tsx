import React from 'react';

import { shallow, configure, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Input from './input.component';
import { TextInput } from 'react-native-gesture-handler';

configure({ adapter: new Adapter() });

describe('input snapshot tests', () => {
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
  beforeEach(() => {
    wrapper = shallow(<Input />);
  });
  it('<Input />: propsなし', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('<Input secureTextEntry />: secureTextEntry->true', () => {
    expect(wrapper.setProps({ secureTextEntry: true })).toMatchSnapshot();
  });
  it('<Input placeholder={"インプット"} />: placeholder->インプット', () => {
    expect(wrapper.setProps({ placeholder: 'インプット' })).toMatchSnapshot();
  });
  it('<Input value={"インプット"} />: value->インプット', () => {
    expect(wrapper.setProps({ value: 'インプット' })).toMatchSnapshot();
  });
  it('<Input error />: error->true', () => {
    expect(wrapper.setProps({ error: true })).toMatchSnapshot();
  });
});

it('input onChangeText', () => {
  const onChangeText = jest.fn();
  const wrapper = shallow(<Input onChangeText={onChangeText} />);
  wrapper.find(TextInput).simulate('changeText', 'インプット');
  expect(onChangeText).toBeCalledWith('インプット');
});
