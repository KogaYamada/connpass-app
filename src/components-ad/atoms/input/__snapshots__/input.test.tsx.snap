// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input snapshot tests <Input />: propsなし 1`] = `
<Component
  onChangeText={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#F2F2F2",
        "borderColor": "#DBDBDB",
        "borderRadius": 7,
        "borderStyle": "solid",
        "borderWidth": 1,
        "fontSize": 18,
        "height": 48,
        "padding": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`input snapshot tests <Input error />: error->true 1`] = `
<Component
  onChangeText={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#F2F2F2",
        "borderColor": "#DBDBDB",
        "borderRadius": 7,
        "borderStyle": "solid",
        "borderWidth": 1,
        "fontSize": 18,
        "height": 48,
        "padding": 10,
      },
      Object {
        "borderColor": "red",
      },
    ]
  }
/>
`;

exports[`input snapshot tests <Input placeholder={"インプット"} />: placeholder->インプット 1`] = `
<Component
  onChangeText={[Function]}
  placeholder="インプット"
  style={
    Array [
      Object {
        "backgroundColor": "#F2F2F2",
        "borderColor": "#DBDBDB",
        "borderRadius": 7,
        "borderStyle": "solid",
        "borderWidth": 1,
        "fontSize": 18,
        "height": 48,
        "padding": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`input snapshot tests <Input secureTextEntry />: secureTextEntry->true 1`] = `
<Component
  onChangeText={[Function]}
  secureTextEntry={true}
  style={
    Array [
      Object {
        "backgroundColor": "#F2F2F2",
        "borderColor": "#DBDBDB",
        "borderRadius": 7,
        "borderStyle": "solid",
        "borderWidth": 1,
        "fontSize": 18,
        "height": 48,
        "padding": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`input snapshot tests <Input value={"インプット"} />: value->インプット 1`] = `
<Component
  onChangeText={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#F2F2F2",
        "borderColor": "#DBDBDB",
        "borderRadius": 7,
        "borderStyle": "solid",
        "borderWidth": 1,
        "fontSize": 18,
        "height": 48,
        "padding": 10,
      },
      undefined,
    ]
  }
  value="インプット"
/>
`;
